yarn-project/end-to-end/src/e2e_client_flows/client_test_flows.ts

```diff
@@ -0,0 +1,298 @@
+import { getSchnorrAccount, getSchnorrWallet } from '@aztec/accounts/schnorr';
+import type { InitialAccountData } from '@aztec/accounts/testing';
+import {
+  type AccountWallet,
+  AztecAddress,
+  type AztecNode,
+  FeeJuicePaymentMethodWithClaim,
+  type Logger,
+  type PXE,
+  createLogger,
+} from '@aztec/aztec.js';
+import { CheatCodes } from '@aztec/aztec.js/testing';
+import { FEE_FUNDING_FOR_TESTER_ACCOUNT } from '@aztec/constants';
+import { type DeployL1ContractsArgs, RollupContract, createL1Clients, deployL1Contract } from '@aztec/ethereum';
+import { ChainMonitor } from '@aztec/ethereum/test';
+import { randomBytes } from '@aztec/foundation/crypto';
+import { EthAddress } from '@aztec/foundation/eth-address';
+import { Fr } from '@aztec/foundation/fields';
+import { TestERC20Abi } from '@aztec/l1-artifacts/TestERC20Abi';
+import { TestERC20Bytecode } from '@aztec/l1-artifacts/TestERC20Bytecode';
+import { CounterContract } from '@aztec/noir-contracts.js/Counter';
+import { FPCContract } from '@aztec/noir-contracts.js/FPC';
+import { FeeJuiceContract } from '@aztec/noir-contracts.js/FeeJuice';
+import { SponsoredFPCContract } from '@aztec/noir-contracts.js/SponsoredFPC';
+import { TokenContract as BananaCoin, TokenContract } from '@aztec/noir-contracts.js/Token';
+import { ProtocolContractAddress } from '@aztec/protocol-contracts';
+import { getCanonicalFeeJuice } from '@aztec/protocol-contracts/fee-juice';
+import { type PXEServiceConfig, createPXEService, getPXEServiceConfig } from '@aztec/pxe/server';
+import { GasSettings } from '@aztec/stdlib/gas';
+import { deriveSigningKey } from '@aztec/stdlib/keys';
+
+import { MNEMONIC } from '../fixtures/fixtures.js';
+import {
+  type ISnapshotManager,
+  type SubsystemsContext,
+  createSnapshotManager,
+  deployAccounts,
+} from '../fixtures/snapshot_manager.js';
+import { mintTokensToPrivate } from '../fixtures/token_utils.js';
+import { type SetupOptions, setupCanonicalFeeJuice } from '../fixtures/utils.js';
+import { CrossChainTestHarness } from '../shared/cross_chain_test_harness.js';
+import { FeeJuicePortalTestingHarnessFactory, type GasBridgingTestHarness } from '../shared/gas_portal_test_harness.js';
+
+const { E2E_DATA_PATH: dataPath } = process.env;
+
+export type AccountType = 'ecdsar1' | 'schnorr';
+
+export class ClientFlowsTest {
+  private snapshotManager: ISnapshotManager;
+
+  public logger: Logger;
+  public pxe!: PXE;
+  public aztecNode!: AztecNode;
+  public cheatCodes!: CheatCodes;
+
+  public adminWallet!: AccountWallet;
+  public adminAddress!: AztecAddress;
+
+  public sequencerAddress!: AztecAddress;
+
+  public coinbase!: EthAddress;
+
+  public gasSettings!: GasSettings;
+
+  public feeJuiceContract!: FeeJuiceContract;
+  public bananaCoin!: BananaCoin;
+  public candyBarCoin!: TokenContract;
+  public bananaFPC!: FPCContract;
+  public sponsoredFPC!: SponsoredFPCContract;
```

Comment:
It'd be nice to have at least brief descriptions of what these are supposed to be, to not make people go insane. Alternatively names such as 'test token' suffice.

Reply:
Done